packages/controllers/src/snaps/SnapController.ts

```diff
@@ -580,26 +580,25 @@ export class SnapController extends BaseController<
         this._maxIdleTime &&
         timeSince(runtime.lastRequest) > this._maxIdleTime
       ) {
-        this._stopSnap(snapId);
+        this._stopSnap(snapId, SnapStatusEvent.stop);
       }
     });
   }
 
-  _onUnresponsiveSnap(snapId: SnapId) {
-    this._transitionSnapState(snapId, SnapStatusEvent.crash);
```

Comment:
The call to `_transitionSnapState` outside of `_stopSnap` here and in `_onUnhandledSnapError` was the crux of the bug.